### PR TITLE
feat(plugins/permissionsViewer) add roles & users context menus to copy id

### PR DESCRIPTION
**Pull request by CatGirlDShadow**
title
image

### CODE REVIEW

File: src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx [hunk 1]
```diff
@@ -200,24 +208,53 @@ function RoleContextMenu({ guild, roleId, onClose }: { guild: Guild; roleId: str
             aria-label="Role Options"
         >
             <Menu.MenuItem
-                id="vc-pw-view-as-role"
-                label="View As Role"
+                id="vc-copy-role-id"
+                label="Copy Id"
```
Comment (Vendicated) [inline src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx:212]:
```suggestion
                label="Copy Role ID"
```

File: src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx [hunk 2]
```diff
@@ -200,24 +208,53 @@ function RoleContextMenu({ guild, roleId, onClose }: { guild: Guild; roleId: str
...
+function UserContextMenu({ userId, onClose }: { userId: string; onClose: () => void; }) {
+    return (
+        <Menu.Menu
+            navId={cl("user-context-menu")}
+            onClose={ContextMenuApi.closeContextMenu}
+            aria-label="User Options"
+        >
+            <Menu.MenuItem
+                id="vc-copy-user-id"
+                label="Copy Id"
```
Comment (Vendicated) [inline src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx:255]:
```suggestion
                label="Copy Role ID"
```

**Review by Vendicated — COMMENTED** [review]
looks great! just two minor changes to make the text consistent with Discord

**Comment by Vendicated** [comment]
please do not uncheck this

**Comment by Vendicated** [comment]
https://github.com/Vendicated/Vencord/pull/2367#issuecomment-2093744498

File: src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx [hunk 3]
```diff
@@ -200,24 +208,53 @@ function RoleContextMenu({ guild, roleId, onClose }: { guild: Guild; roleId: str
...
+function UserContextMenu({ userId, onClose }: { userId: string; onClose: () => void; }) {
+    return (
+        <Menu.Menu
+            navId={cl("user-context-menu")}
+            onClose={ContextMenuApi.closeContextMenu}
+            aria-label="User Options"
+        >
+            <Menu.MenuItem
+                id="vc-copy-user-id"
+                label="Copy Role ID"
```
Comment (Vendicated) [inline src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx:255]:
```suggestion
                label="Copy User ID"
```

**Comment by Vendicated** [comment]
https://github.com/Vendicated/Vencord/pull/2367#issuecomment-2093744498

File: src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx (hunk 3 above)
Comment (Cynosphere) [reply src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx:255]:
```suggestion
                label={i18n.Messages.COPY_ID_USER}
```

Comment (Vendicated) [reply src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx:255]: true thats even better

Comment (CatGirlDShadow) [reply src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx]: done

**Comment by CatGirlDShadow** [comment]
> [#2367 (comment)](https://github.com/Vendicated/Vencord/pull/2367#issuecomment-2093744498)

How do i change it now?

**Pull request closed without merging**